addons/udes_stock_cron/models/stock_picking.py

```diff
@@ -265,9 +268,47 @@ def _process_pickings(self):
         if unsatisfied:
             # Unreserve if the picking type cannot handle partials or it
             # can but there is nothing allocated (no stock.move.lines)
-            if not self.picking_type_id.u_handle_partials or not mls:
+            # If there are several picking types, they should have all same handle_partials,
+            # can be improved to handle it better if necessary.
+            handle_partial = self.picking_type_id[0].u_handle_partials
+            if not handle_partial or not mls:
                 # construct error message, report only products
                 # that are unreservable.
                 moves = unsatisfied.move_lines.filtered(unsatisfied_state)
                 self.raise_insufficiency_error(moves)
         return {"newly_processed": newly_processed, "mls": mls}
+
+    def add_more_pickings_to_reserve(self, picking_type):
+        """
+        Checking different configuration of the picking type, and add more pickings to reserve when applied.
+        Method can be override on specific implementations to extend pickings to reserve.
+
+        Args:
+            picking_type (recordset): Stock picking type recordset.
+        Returns:
+            pickings (recordset): Picking to reserve recordset
+        """
+        pickings = self
+        if self.batch_id:
+
+            if self.u_reserve_batches:
+                pickings = pickings.batch_id.picking_ids
+            if picking_type.u_reserve_batches:
+                # Process the batches pickings
+                pickings = pickings.batch_id.picking_ids
+
+        return pickings
+
+    def add_processed_pickings(self, processed_pickings):
+        """
+        Checking different configuration of the picking type, and add processed pickings to be skipped when reserving.
+        Method can be override on specific implementations to extend pickings that are skipped.
+
+        Args:
+            processed_pickings  (recordset): Stock picking recordset.
+        Returns:
+            processed_pickings  (recordset): Stock picking recordset.
+        """
+        if self.batch_id and self.batch_id.state == "draft":
```

Comment:
nit: if `self.batch_id` is emptyset, then `self.batch_id.state` will equal False. I don't think we need check for `if self.batch_id and`